src/clj/game/cards/hardware.clj

```diff
@@ -905,10 +905,15 @@
                                       (effect-completed state side eid)))}}}]})
 
 (define-card "Hippo"
+  (letfn [(build-hippo-pred [outermost-ices]
+    (fn [events]
+      (not (empty? (filter #(true? %) (map #(and (same-card? % (first events))
```

Comment:
Also this `not empty filter` seems bad, but at least it works. I've tried to use `or` here and it didn't work with `map` result.

Reply:
Whoops, I missed this. For future reference, instead of `(not (empty (filter`, you can write `(seq (filter` or `(not-empty (filter`. Additionally, you don't have to wrap single-argument functions (predicates), you can just use it directly: `(filter true? (map ...`. This still isn't quite ideal because we want to be able to return truth-y things, not just `true`, so a better way would be `(filter identity` to remove false-y values.